adaptdl/adaptdl/torch/gradient_noise_scale.py

```diff
@@ -53,6 +54,9 @@ def __init__(self, adp, optimizer,
                               else torch.distributed.get_world_size())
         self._accum_scale = accum_scale or self._num_replicas
         self._prev_grads = None
+        self._is_adam = optimizer.__class__.__name__ in ['Adam', 'AdamW']
```

Comment:
I think it would be cleaner to use a new subclass with a different `_calculate_preconditioner` function, where the base `GradientNoiseScale._calculate_preconditioner` returns the identity preconditioner.